blk/components/virt.c

```diff
@@ -175,18 +175,15 @@ static bool handle_mbr_reply()
 
     microkit_arm_vspace_data_invalidate(mbr_req_data.drv_addr,
                                         mbr_req_data.drv_addr + (BLK_TRANSFER_SIZE * mbr_req_data.count));
-    memcpy(&msdos_mbr, (void *)mbr_req_data.drv_addr, sizeof(struct msdos_mbr));
+    sddf_memcpy(&msdos_mbr, (void *)mbr_req_data.drv_addr, sizeof(struct msdos_mbr));
     fsmalloc_free(&fsmalloc, mbr_req_data.drv_addr, mbr_req_data.count);
 
     return true;
 }
 
 void init(void)
 {
-    // @ericc: Hack, spin wait for config from driver to be set
-    while (!(((blk_storage_info_t *)blk_config_driver)->ready)) {
-        asm("");
-    }
+    while (!__atomic_load_n(&((blk_storage_info_t *)blk_config_driver)->ready, __ATOMIC_ACQUIRE));
```

Comment:
The trailing semicolon gets lost; I'd prefer it as a line on its own.
As before, we should refactor this to get rid of all the casts.

Reply:
Fixed